src/org/exist/client/InteractiveClient.java

```diff
@@ -2213,23 +2086,22 @@ private void connectToDatabase() {
         try {
             connect();
         } catch (final Exception cnf) {
-            if (startGUI && frame != null) {
+            if (options.startGUI && frame != null) {
                 frame.setStatus("Connection to database failed; message: " + cnf.getMessage());
             } else {
                 System.err.println("Connection to database failed; message: " + cnf.getMessage());
             }
             cnf.printStackTrace();
-            System.exit(1);
+            System.exit(SystemExitCodes.CATCH_ALL_GENERAL_ERROR_EXIT_CODE);
         }
     }
 
     /**
      * Main processing method for the InteractiveClient object
      *
-     * @param args arguments from main()
      * @return true on success, false on failure
      */
-    public boolean run(final String args[]) throws Exception {
+    public boolean run() throws Exception {
```

Comment:
well things moved from method to constructor. Since the parameters are from a different origin,  the API changes. Not sure if it is a problem for e.g. the webstart client?